### PR TITLE
Updates for confused part of code which can cause a problems

### PR DESCRIPTION
**Pull request by entymon**
Updates for confused part of code which can cause a problems during using. 

1. Move common config to top as its empty and override previous configs
2. Remove env from provider because there is not place for custom variables

### CODE REVIEW

File: serverless.yml [hunk 1]
```diff
@@ -19,12 +19,10 @@ provider:
   stage: ${opt:stage, 'sandbox'}
   iamRoleStatements:
     ${file(config/role-statements/common.yml)}
-  environment:
-    ${file(config/environments/${opt:stage, 'sandbox'}.yml)}
```
Comment (entymon) [inline serverless.yml:23]: This is not a place for this variable. BTW this part is duplicated as was added to custom.

File: serverless.yml [hunk 2]
```diff
@@ -19,12 +19,10 @@ provider:
...
 
 functions:
   - ${file(config/functions/hello.yml)}
 
 resources:
-  - ${file(config/resources/api-gateway.yml)}
```
Comment (mgarciadelojo) [inline serverless.yml:29]: I would not move this line since the files here were sorted alphabetically

File: serverless.yml [hunk 3]
```diff
@@ -19,12 +19,10 @@ provider:
...
   - ${file(config/resources/common.yml)}
+  - ${file(config/resources/api-gateway.yml)}
```
Comment (entymon) [inline serverless.yml:28]: common.yml moved to top as the content of the file is empty and the empty value of Resources in this file overrides not empty values for previous files. Tested with latest Serverless Framework.

File: serverless.yml (hunk 2 above)
Comment (entymon) [reply serverless.yml:29]: Hey @mgarciadelojo I left the comment why I do this :)

File: serverless.yml (hunk 3 above)
Comment (mgarciadelojo) [reply serverless.yml:28]: Let's remove common.yml then, it seems not to make sense to have it here for now

**Pull request merged**